### DISCUSSION TITLE
Anyone was able to run it succesfully?

### DISCUSSION THREAD

**Discussion by yarodevuci** [xai-org/grok-1#168] [General]
I want to know what is look like when running? What you can do with it? Checkpoints are not finetuned or anything.. so what to do next with this code and trained data?

**Comment by LagPixelLOL** [comment]
ran it on 8x a100 80g with the code in this repo(no modification, i just added a loop to get input from terminal), using 524GB of vram during single batch inference with nearly no context(10~100 tokens input), speed is only 7 tokens per second ish
even tho they said its a base model, but funny it can get into a role of assistant pretty well with just `User: xxx\n\nAssistant:` format, which isnt like how llama 2 base model behaved, still the dialogue ability is very very bad, so i think for this model to be useful it still needs more fine tunes
also if asked what the knowledge cut off is, it will say 2021/9, and sometimes it thinks it's made by openai, idk if this is because of the sheer amount of shared chatgpt chatlog on the internet or its because they trained grok on openai's models' outputs

**Reply by penut85420** [reply]
~~Are you running in 16 bit or 8 bit?~~

Is there only 8-bit weights available at the moment?

**Reply by Zblocker64** [reply]
What was the other specs of the server you ran it on?

**Reply by LagPixelLOL** [reply]
@penut85420 seems like there's only 8bit version

**Reply by LagPixelLOL** [reply]
@Zblocker64 1.5TB ram and 2x 8480+

**Reply by Zblocker64** [reply]
@LagPixelLOL is the 1.5TH of ram necessary?

**Reply by LagPixelLOL** [reply]
@Zblocker64 no, just have larger ram than the model itself, but i think even this isnt needed

**Reply by ghost** [reply]
Hi @LagPixelLOL , Congrats on running it!
Are you running it on the host/VM or as container (Kubernetes pod)?

I've also managed to run it on the Host and even in K8s Pod (but with the grok-1 mounted from the host over `ext4` as it fails most of the times with `overlay` FS -- details [here](https://github.com/xai-org/grok-1/issues/164#issuecomment-2007077866)).
Could you please share the updated code with the loop to get input from terminal? :pray:

**Reply by LagPixelLOL** [reply]
@andy108369 i just used python's input function, very easy to do
im running it on host, not container
id like to do more experiments but im really not familiar with jax, need to port it to pytorch somehow

**Reply by yarodevuci** [reply]
@LagPixelLOL what does it look like when running ?

**Reply by LagPixelLOL** [reply]
@yarodevuci you mean performance?

**Reply by AdaptiveStep** [reply]
Any reports on the token context size? Can it handle 200k+ tokens? In that case its broken world record.

**Reply by yarodevuci** [reply]
I mean the actual output in the console. how to use this trained data once it's running.

**Reply by ghost** [reply]
@yarodevuci you can see its console output on some of the screenshots and recordings I've shared here https://github.com/xai-org/grok-1/issues/164#issuecomment-2007077866

The issue is that it doesn't really finish the ouput and prematurely exits.

**Reply by ghost** [reply]
@LagPixelLOL 
> and 2x 8480+

It appears that it needs at least 8 GPUs, 
https://github.com/xai-org/grok-1/discussions/183

Did you modify the python code for it to work with 2 GPUs?

> i just used python's input function, very easy to do

Could you please share your code?

> im running it on host, not container

Even when I'm running it on the host, it just prematurely exits (exit code 0) as it types the response 🤷‍♂️
https://github.com/xai-org/grok-1/issues/164#issuecomment-2007077866

**Reply by LagPixelLOL** [reply]
@andy108369 8480+ is the CPU, the GPU is 8x A100 80GB
I already implemented it in PyTorch, I'll finish the sampling in PyTorch then I'll commit it to my fork.
https://github.com/LagPixelLOL/grok-1-pytorch

**Reply by iayub02** [reply]
@LagPixelLOL We are trying with RunPod Pytorch 2.1, 8 x A100 SXM 80GB and 256 vCPU 2002 GB RAM.

**Reply by iayub02** [reply]
What kind of performance are you getting now on PyTorch?

**Reply by LagPixelLOL** [reply]
@iayub02 I just found someone who made a full huggingface transformers implementation and it's way better than mine, I'll test on this instead https://huggingface.co/keyfan/grok-1-hf

**Reply by Cpmcharlesandrew** [reply]
pip install dm-haiku
pip install jax jaxlib
On Windows Run the following first to install the prerequisites.

**Reply by iamdeepakgit** [reply]
Yes it's running good and prefect 😉

**Comment by yarodevuci** [comment]
Has anyone tried on Mac M3 with 128GB shared memory?

**Reply by MrRaja23** [reply]
Assuming it needs 500gb vram in its current state that would go the same for ram no?

**Comment by dominiquegarmier** [comment]
@LagPixelLOL 
> need to port it to pytorch somehow

I created a pytorch implementation: https://github.com/DominiqueGarmier/grok-pytorch, you can't really use it at this point but maybe it would be a good starting point.

**Reply by LagPixelLOL** [reply]
@dominiquegarmier oh wow i made one myself too! its in https://github.com/LagPixelLOL/grok-1-pytorch

**Reply by LagPixelLOL** [reply]
apparently someone already made a full huggingface implementation of grok, here: https://huggingface.co/keyfan/grok-1-hf

**Reply by yarodevuci** [reply]
will it work on one video card?

**Reply by dominiquegarmier** [reply]
> will it work on one video card?

no, not even maybe

**Reply by dominiquegarmier** [reply]
> apparently someone already made a full huggingface implementation of grok, here: https://huggingface.co/keyfan/grok-1-hf

hmm I looked at the code and it appears to just be copy of Mixtral with "Mixtral" replaced with "Grok", but maybe they edited some things to make it compatible.

**Reply by LagPixelLOL** [reply]
@dominiquegarmier Yeah but they did modify some of the code to adapt it to Grok-1, I'm currently trying to convert it to safentensors ~~but IDK why it's failing~~ nvm figured it out.

**Comment by ghost** [comment]
I finally was able to run it (though PyTorch version of it https://huggingface.co/hpcai-tech/grok-1 ) on Akash Network.
Details https://github.com/akash-network/awesome-akash/pull/507#issuecomment-2015528500

![image](https://github.com/xai-org/grok-1/assets/107317698/70909402-a77b-41cd-bcc6-21b9a571344a)

**Comment by iayub02** [comment]
Is it consuming all 8 GPU when you ask a question?  We got it working but
it seems super inefficient.

George

**Comment by ateeq-pk** [comment]
I have made the model successfully working on 8 A100 80GB, but the response from model was not satisfactory. I am going to try to do second time and more properly set query.

**Instructions:** [https://github.com/xai-org/grok-1/discussions/246](https://github.com/xai-org/grok-1/discussions/246)

**Comment by yh-yao** [comment]
https://www.fedml.ai/models/941?owner=xai-org Try here. I just use it for the completion task. The current Grok-1 does not support chat template. It is much better now. 
![7c50aa67d76e510fabc659a0b31529](https://github.com/xai-org/grok-1/assets/24760321/62e15fd0-cc9b-48f6-916a-082ff56e6a38)

**Comment by Xmabel112Lubi** [comment]
Hello can me someone explain what is that all thanks

Odoslané z aplikácie Outlook pre Android<https://aka.ms/AAb9ysg>
________________________________
